alvr/audio/src/lib.rs

```diff
@@ -525,8 +525,8 @@ pub fn receive_samples_loop(
                 let volume = f as f32 / batch_frames_count as f32;
                 for c in 0..channels_count {
                     let index = f * channels_count + c;
-                    sample_buffer_ref[index] =
-                        sample_buffer_ref[index] * volume + drained_samples[index] * (1. - volume);
+                    sample_buffer_ref[index] = sample_buffer_ref[index]
+                        .mul_add(volume, drained_samples[index] * (1. - volume));
```

Comment:
Unless there is a clear benefit on using `.mul_add()`, i'm not a fan of it because it makes the code harder to read. I would expect the compiler to make this kind of optimization anyway for us.

Reply:
Well it actually can't, that's why there's a lint for that sorta stuff in the first place, since two step does two float operations and the fused operation only does one you end up with more precision overall which technically changes the value so the compiler can't actually optimize that. Theoretically this is faster and more precise but it also doesn't really matter overall so I'm open to not doing it especially as a lint.

Reply:
We haven't measured the performance of audio code, so I would say this is a premature optimization. for video code we did actually measure performance and did informed optimizations

Reply:
The way the function is used, half infix and half prefix, is rather weird. My preference rule is that function operators should always be displayed as a prefix (meaning that all operands follow the function name, even when one of them is "self"). For operators that have an ordering for operands, i would always prefer using the symbols operators, displayed infix.